sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -222,61 +222,40 @@ def __internal_api__get_items_as_dataclasses__(
     ) -> List[dataset_item.DatasetItem]:
         results: List[dataset_item.DatasetItem] = []
 
-        if dataset_item_ids is not None:
-            dataset_item_ids = set(dataset_item_ids)  # type: ignore
-
-        while True:
-            stream = self._rest_client.datasets.stream_dataset_items(
-                dataset_name=self._name,
-                last_retrieved_id=results[-1].id if len(results) > 0 else None,
-            )
-
-            previous_results_size = len(results)
-            if nb_samples is not None and len(results) == nb_samples:
-                break
-
-            item_bytes = b"".join(stream)
-            for line in item_bytes.split(b"\n"):
-                if len(line) == 0:
-                    continue
-
-                full_item_content: Dict[str, Any] = json.loads(
-                    line.decode("utf-8").strip()
-                )
-                data_item_content = (
-                    full_item_content["data"] if "data" in full_item_content else {}
-                )
-
-                dataset_item_id = full_item_content.get("id")
-
-                if dataset_item_ids is not None:
-                    if dataset_item_id not in dataset_item_ids:
-                        continue
-                    else:
-                        dataset_item_ids.remove(dataset_item_id)
+        dataset_item_ids_set = (
+            set(dataset_item_ids) if dataset_item_ids is not None else None
+        )
 
-                item = dataset_item.DatasetItem(
-                    id=full_item_content.get("id"),  # type: ignore
-                    trace_id=full_item_content.get("trace_id"),  # type: ignore
-                    span_id=full_item_content.get("span_id"),  # type: ignore
-                    source=full_item_content.get("source"),  # type: ignore
-                    **data_item_content,
-                )
+        dataset_items = rest_stream_parser.read_and_parse_stream(
+            stream=self._rest_client.datasets.stream_dataset_items(
```

Comment:
This call might be needed to be made multiple times, one stream has a limit of 2k items in it, so, there should be a while loop similar to the original code in main branch (it's also a reason why dataset.py should be responsible for keeping only the items with the correct id, not the rest_and_parse_stream function)

Reply:
The 2k limit clarifies it!

Updated back to while loop